modules/database/funcs.go

```diff
@@ -229,3 +231,34 @@ func (d *Database) getDBWithName(
 
 	return nil
 }
+
+//
+// Returns a *Database object with specified name and namespace
+//
+func GetDatabaseByName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+) *Database {
```

Comment:
as `getDBWithName` might fail `GetDatabaseByName` can fail too. So you need to return (*Database, error) to be able to signal failure to the caller too.

Reply:
ok